sale_global_discount/models/sale_order.py

```diff
@@ -42,10 +42,34 @@ def get_discounted_global(self, price=0, discounts=None):
         price *= 1 - (discount / 100)
         return self.get_discounted_global(price, discounts)
 
+    def _check_global_discounts_sanity(self):
+        """Perform a sanity check for discarding cases that will lead to
+        incorrect data in discounts.
+        """
+        self.ensure_one()
+        if not self.global_discount_ids:
+            return True
+        taxes_keys = {}
+        for line in self.order_line:
+            if not line.tax_id:
+                raise exceptions.UserError(_(
+                    "With global discounts, taxes in lines are required."
+                ))
+            for key in taxes_keys:
+                if key == line.tax_id:
+                    break
+                elif key & line.tax_id:
```

Comment:
why is this bitwise operation needed to perform the check?

Reply:
You have approved the same operation in the other PR :stuck_out_tongue: 

The idea is to constraint non valid taxes combinations that will lead to incorrect global discount. Example:

- Line 1: tax 1 and tax 2
- Line 2: tax 2

As (tax 1, tax 2) is not the same as (tax 2), first condition doesn't break the loop, but the overlapping of both (the `&` operation) gives you a common tax, raising the constraint.

Reply:
Interesting technique :)